Implements being able to fill bottles with water.

IvanCraft623 opened the pull request:
## Introduction
<!-- Explain existing problems or why this pull request is necessary -->
Players can now fill empty bottles with water like Vanilla.

### Relevant issues
<!-- List relevant issues here -->
N/A

## Changes
### API changes
<!-- Any additions to the API that should be documented in release notes? -->
N/A

## Backwards compatibility
<!-- Any possible backwards incompatible changes? How are they solved, or how can they be solved? -->
Backward Compatible.

## Follow-up
<!-- Suggest any actions to be done before/after merging this pull request -->
<!--
N/A


## Tests
<!--
Details should be provided of tests done. Simply saying "tested" or equivalent is not acceptable.

Attach scripts or actions to test this pull request, as well as the result
-->
- Interacting with water with a bottle subtracts an empty bottle from the player and adds one with water.
- If the player has a full inventory the water bottle will be dropped in front of the player.

IvanCraft623 committed:
Fill bottles!

dktapps commented on src/item/GlassBottle.php:
```diff
@@ -23,6 +23,32 @@
+		if($blockClicked->getId() === BlockLegacyIds::FLOWING_WATER){
+			$watterPotion = VanillaItems::WATER_POTION();
+			if($player->getInventory()->canAddItem($watterPotion)){
+				$player->getInventory()->addItem($watterPotion);
```
This can be simplified by using the results of `addItem()` (it will return the items that didn't fit into the inventory).

dktapps commented on src/item/GlassBottle.php:
```diff
@@ -23,6 +23,32 @@
 
+	public function onInteractBlock(Player $player, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector) : ItemUseResult{
+		if($blockClicked->getId() === BlockLegacyIds::FLOWING_WATER){
+			$watterPotion = VanillaItems::WATER_POTION();
```
```suggestion
			$waterPotion = VanillaItems::WATER_POTION();
```

dktapps commented on src/item/GlassBottle.php:
```diff
@@ -23,6 +23,32 @@
 class GlassBottle extends Item{
 
+	public function onInteractBlock(Player $player, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector) : ItemUseResult{
+		if($blockClicked->getId() === BlockLegacyIds::FLOWING_WATER){
```
It may not be flowing water. Water most commonly exists in its still form in vanilla.

IvanCraft623 committed:
A rewrite

- Now the Water instance is checked instead of the id.
- Fix "watter" typo.
- The result of addItem() is now used instead of canAddItem().
- Fix that water potion was not assigned to the correct slot when the player only has one bottle and has finite resources.

IvanCraft623 committed:
Update GlassBottle.php

dktapps commented:
Implemented on `next-major` by 81edb1bed494639f512427bd97cf481e6f777fd8.
